### PR TITLE
Feature chinedu profile update

### PR DESCRIPTION
**Pull request by ChineduOguejiofor**
#### What does this PR achieve?
- Updates the logic to add User and Vendor profile
- Update logic to add and get order
#### Relevant Trello Tickets

- [Ticket 1](some.com)

### CODE REVIEW

File: functions/controllers/order.js [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
 
 exports.addOrder = async (req, res, next) => {
     const {
-        vendorID, vendorLocationID, orderItems, orderDetails, totalPrice,
+        vendorID, address, orderItems, orderDetails, totalPrice,
     } = req.body;
-    const { userID } = req.authId;
-    const orderID = randomID();
-    const orderStatus = 'waiting';
+    if (!vendorID || !address || !orderItems || !orderDetails || !totalPrice) {
```
Comment (Way2nnadi) [inline functions/controllers/order.js]: save this conditional statement to a verbose variable (i.e. `hasMissingInfo` or something like this
)

File: functions/controllers/order.js [hunk 2]
```diff
@@ -1,22 +1,31 @@
...
+        return res.status(BAD_REQUEST).json({
+            status: 'fail',
+            data: { title: 'Please pass in the necessary parameters' },
```
Comment (Way2nnadi) [inline functions/controllers/order.js]: why are you sending data on a bad request. just send a message and use `http-status-code` to get the message that corresponds to `BAD_REQUEST`

File: functions/controllers/order.js [hunk 3]
```diff
@@ -1,22 +1,31 @@
...
+        });
+    }
+    const userID = req.authId;
+    const orderID = randomString(20);
```
Comment (Way2nnadi) [inline functions/controllers/order.js]: Do not roll out you own solution for Ids. Use the [uuid](https://www.npmjs.com/package/uuid) npm package to generate your ids

File: functions/controllers/order.js [hunk 4]
```diff
@@ -31,34 +40,38 @@ exports.addOrder = async (req, res, next) => {
...
 
-exports.orderList = async (req, res, next) => {
-    const { userID } = req.authId;
+exports.getOrders = async (req, res, next) => {
+    const userID = req.authId;
+    const { userType } = req.body;
+    if (!userType) {
+        return res.status(BAD_REQUEST).json({
+            status: 'fail',
+            data: { title: 'Please provide the User Type' },
```
Comment (Way2nnadi) [inline functions/controllers/order.js]: see comment fomr above

File: functions/controllers/order.js [hunk 5]
```diff
@@ -31,34 +40,38 @@ exports.addOrder = async (req, res, next) => {
...
+        });
+    }
+    let ordersSnapshot;
+    if (userType === 'user') {
+        ordersSnapshot = await db.collection('orders').where('userID', '==', userID);
+    }
+    if (userType === 'vendor') {
+        ordersSnapshot = await db.collection('orders').where('vendorID', '==', userID);
+    }
```
Comment (Way2nnadi) [inline functions/controllers/order.js]: make line 66-71 and `if/else if` statement.

File: functions/controllers/profile.js [hunk 6]
```diff
@@ -1,35 +1,87 @@
...
+
 exports.updateProfile = async (req, res, next) => {
     const { info, addresses, userType } = req.body;
     const userID = req.authId;
-    if (!info || !addresses || !userType) {
+    if (!userType) {
         return res.status(BAD_REQUEST).json({
             status: 'fail',
-            data: { title: 'Please provide the necessary parameters' },
+            data: { title: 'Please provide the User Type' },
```
Comment (Way2nnadi) [inline functions/controllers/profile.js]: see above comment

File: functions/utils/index.js [hunk 7]
```diff
@@ -1,5 +1,9 @@
...
\ No newline at end of file
+const randomString = length => {
+    const chars = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+    let result = '';
+    for (let i = length; i > 0; --i) result += chars[Math.floor(Math.random() * chars.length)];
+    return result;
+};
+
+module.exports = randomString;
+
```
Comment (Way2nnadi) [inline functions/utils/index.js]: delete this function

File: functions/controllers/order.js (hunk 1 above)
Comment (ChineduOguejiofor) [reply functions/controllers/order.js]: I extracted this to use Validator JS

File: functions/controllers/order.js (hunk 2 above)
Comment (ChineduOguejiofor) [reply functions/controllers/order.js]: Removed.
Replaced with the validation middleware

File: functions/controllers/order.js (hunk 3 above)
Comment (ChineduOguejiofor) [reply functions/controllers/order.js]: Removed
Replaced with uuid
It's even way easier that way. Nice Idea
Thanks

File: functions/controllers/order.js (hunk 4 above)
Comment (ChineduOguejiofor) [reply functions/controllers/order.js]: Removed

File: functions/controllers/order.js (hunk 5 above)
Comment (ChineduOguejiofor) [reply functions/controllers/order.js]: Done

File: functions/controllers/profile.js (hunk 6 above)
Comment (ChineduOguejiofor) [reply functions/controllers/profile.js]: Replaced with the validator js equivalent

File: functions/middleware/order.js [hunk 8]
```diff
@@ -1,12 +1,16 @@
...
+
 const { BAD_REQUEST } = HttpStatus;
-exports.orderValidation =  (req, res, next) => {
-    const { vendorID, vendorLocation, orderItems, orderDetails, totalPrice } = req.body;
-    if(!vendorID && !vendorLocation && !orderItems && !orderDetails && totalPrice ) {
+exports.orderValidation = (req, res, next) => {
+    const {
+        vendorID, vendorLocation, orderItems, orderDetails, totalPrice,
+    } = req.body;
+    if (!vendorID && !vendorLocation && !orderItems && !orderDetails && totalPrice) {
```
[the remark this answers is not in the document]
Comment (ChineduOguejiofor) [reply functions/middleware/order.js]: Removed

File: functions/utils/index.js (hunk 7 above)
Comment (ChineduOguejiofor) [reply functions/utils/index.js]: Deleted
Thanks

**Comment by ChineduOguejiofor** [comment]
All comments have been resolved
Medium level refactor implemented

**Pull request merged**